bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/composite/CompositeModuleHandlerFactory.java

```diff
@@ -78,87 +78,102 @@ public CompositeModuleHandlerFactory(BundleContext bc, ModuleTypeManager mtManag
 
     @SuppressWarnings({ "unchecked" })
     @Override
-    public void ungetHandler(Module module, String ruleUID, ModuleHandler handler) {
-        ModuleHandler handlerOfModule = handlers.get(ruleUID + module.getId());
+    public void ungetHandler(Module module, String childModulePrefix, ModuleHandler handler) {
+        ModuleHandler handlerOfModule = handlers.get(childModulePrefix + module.getId());
```

Comment:
In the other places, you put a colon between the prefix and the ID. Why it is missing here?